vit/application.py

```diff
@@ -195,11 +198,51 @@ def _task_attribute_replace(task, attribute):
                 else:
                     return str(task[attribute])
             return ''
+
+        # currently active filter lookup functions. all of these return the
+        # internally used filter representation, which is a flat list of strings
+        _extra_filter = lambda: self.extra_filters
+        _report_filter = lambda: self.model.active_report_filter()
+        _context_filter = lambda: self.active_context_filter()
+
+        # _view_filter is simply a concat of these three filters. logical 'and' is
+        # implied by taskwarrior, just need to make sure to preserve the internal
+        # precedence within each of the three sub-filters
+        def _protect_precedence(filter):
+            return ['(', *filter, ')'] if len(filter) > 1 else filter
+        def _view_filter():
+            parts = [ _extra_filter(), _report_filter(), _context_filter() ]
+            # filter out empty parts
+            parts = [ part for part in parts if part ]
+            # if there is more than one non-empty filter, and at least one part
+            # potentially has complex internal precedence structure...
+            if len(parts) > 1 and any([ len(part) > 1 for part in parts ]):
```

Comment:
This seems like it might be overkill -- is there any reason why we can't always `_protect_precedence()`?

Reply:
The point is to avoid adding parentheses which would lead to really cluttered-looking replacements that are not necessary. E.g. if there is currently only one filter (say the report has `status:PENDING or +next`, no other filters) then the `{VIEW_FILTER}` would unnecessarily resolve to `( status:PENDING or +next )`, running the same binding again would lead to `( ( status:PENDING or +next ) )` and so on. The current solution is not perfect, but it does the job for 90% of cases with one inexpensive line of code.

I was thinking about making this a dedicated helper function as it might be useful in other situations as well, but wasn't sure about naming convention so didn't do it. How would you feel about having a method such as the following as part of the Application class, or is there a file where small static helper functions can go?
```python
# argument is a list of flat lists, returns one flat list expression which preserves the
# internal precedence that any of the filters might have
def merge_filters(filters):
  ...
```

Reply:
`util.py` already exists, and seems like a reasonable location for it.